src/main.dart

```diff
@@ -1,8 +1,118 @@
-// TODO: Run task list app
-void main(List<String> args) {
-  print('Hello World');
+import 'dart:io';
+
+class Task {
+  int taskID;
+  String taskName;
+  String description;
+  bool isCompleted;
+  
+  Task(this.taskID, this.taskName, this.description, this.isCompleted);
 }
 
-class TaskListApp {
-  // TODO: Implement Task list app
+ class Funcions {
+
+List<Task> tasks = [];
+
+int taskID = 1;
+//addTask
+ void addTask(String taskName, String description) {
+tasks.add(Task(taskID,taskName,description, false));
+taskID++;
 }
+
+//removeTask
+void removeTask(int taskID) {
+ int index = tasks.indexWhere((task) => task.taskID == taskID);
+ if (index != -1){
+  print("Task $taskID Removed");
+  tasks.removeAt(index);
+ }
+else{
+  print("Task $taskID not found");
+}}
+
+//Complete Task
+void CompleteTask(int taskID) {
+  var task = tasks.firstWhere((task) => task.taskID == taskID);
+   if (task != null ) {
+    task.isCompleted = true;
+    print("Task $taskID is Completed");
+   } else {
+    print("Task ID $taskID Not found");     
+}}
+
+//displayTask
+void displayTask() {
+for (var i = 0; i < tasks.length; i++) {
+  Task task = tasks[i];  
+  print(
+    "${task.taskID}. [${task.isCompleted ? 'X' : ' '}] ${task.taskName}: ${task.description}");
+}}}
+
+void main() {
+
+Funcions funcions = Funcions();
+
+while(true){
+
+ print("Choose a function: ");
+ print("1) Add Task ");
+ print("2) Remove Task ");
+ print("3) Complete Task ");
+ print("4) Display Task");
+ print("5) Exit");
+print("Enter yor Choice: ");
+
+
+
+   int ? choice = int.tryParse(stdin.readLineSync() ?? "you did't pick a choice ") ;
+  if (choice != "") {
```

Comment:
When parsing an integer, you should check if the result is null before using it instead of checking if it's an empty string.

if (choice == null) {
      print('Invalid input. Please enter a number between 1 and 4.');
      continue;
}